### PR TITLE
Make Geometry values display nicely in the CLI

### PR DESCRIPTION
**Pull request by mbasmanova**
> presto> select st_point(1.2, 3.4);
> select st_point(1.2, 3.4);
>      _col0      
> -----------------
>  POINT (1.2 3.4) 
> (1 row)

Fixes https://github.com/prestodb/presto/issues/9193

### CODE REVIEW

**Comment by findepi** [comment]
@mbasmanova what does `wkt` means in the printed output?

**Comment by electrum** [comment]
This seems strange to mix JSON with text formatting. And it’s actually not legal JSON.

**Comment by mbasmanova** [comment]
@findepi , wkt stands for Well-Known-Text: https://en.wikipedia.org/wiki/Well-known_text

**Comment by mbasmanova** [comment]
@electrum , originally, I wanted to use WKT string as is. E.g. print "POINT (1 2)" instead of "{wkt=POINT (1 2)}". I thought all I need is to annotate toString() method with @JsonVaue. However, this didn't work, because, com.facebook.presto.client.QueryResults#fixValue expects the string to be base64 encoded.

```
            default:
                // for now we assume that only the explicit types above are passed
                // as a plain text and everything else is base64 encoded binary
                if (value instanceof String) {
                    return Base64.getDecoder().decode((String) value);
                }
                return value;
```

Then, I modified toString() method to return base64 encoded WKT. This worked, but CLI printed WKT as an array of bytes...not what I wanted.

Hence, I went with @JsonProperty. 

Do you have any idea what else I could try?

**Comment by mbasmanova** [comment]
cc: @dain

**Comment by electrum** [comment]
I think we need to fix the client protocol to indicate if the representation is binary. We can do this in a backwards compatible way (that new clients will understand).

Returning broken JSON because it happens to work now in the CLI is a bad idea and much harder to fix later.

**Comment by mbasmanova** [comment]
@electrum , are you suggesting to add a boolean flag to server's response and modify the client to apply "return Base64.getDecoder().decode((String) value);" in QueryResults based on the values of the boolean? 

I tried commenting out "return Base64.getDecoder().decode((String) value);" in QueryResults and adding @JsonValue to GeometryWrapper.toString() and got nice results>

> presto> select st_point(1,2);
> select st_point(1,2);
>    _col0    
> -------------
>  POINT (1 2) 
> (1 row)

File: presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java [hunk 1]
```diff
@@ -38,6 +42,18 @@ protected void registerFunctions()
...
     }
 
+    @Test
+    public void testGeometryGetObjectValue()
+            throws Exception
+    {
+        Slice slice = GeoFunctions.stPoint(1.2, 3.4);
+        int[] offsets = {0, slice.length()};
+        boolean[] valueIsNull = {false};
+        VariableWidthBlock block = new VariableWidthBlock(1, slice, offsets, valueIsNull);
```
Comment (electrum) [inline presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java]: Creating a block manually is generally a bad idea, even in a test. The `Type` should be used to create the block:
```java
BlockBuilder builder = GEOMETRY.createBlockBuilder(new BlockBuilderStatus(), 1);
GEOMETRY.writeSlice(builder, slice);
Block block = block.build();

Comment (electrum) [reply presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java]: The contract between blocks and types is that the type owns the block and representation of the data, so only the type should read or write it.

**Comment by mbasmanova** [comment]
Updated test to use GeometryType to create a Block.

File: presto-client/src/main/java/com/facebook/presto/client/QueryResults.java [hunk 2]
```diff
@@ -329,6 +329,7 @@ private static Object fixValue(TypeSignature signature, Object value)
             case IPADDRESS:
             case DECIMAL:
             case CHAR:
+            case "Geometry":
```
Comment (martint) [inline presto-client/src/main/java/com/facebook/presto/client/QueryResults.java]: Could there be case-sensitivity issues?

Comment (mbasmanova) [reply presto-client/src/main/java/com/facebook/presto/client/QueryResults.java]: Given that all over constants are case-sensitive as well, any issues of this kind would affect all these other types as well. I don't think this new constant is any different. Is there something I'm not thinking about?

Comment (martint) [reply presto-client/src/main/java/com/facebook/presto/client/QueryResults.java]: Based on our offline conversation, this should be ok. The only thing I'd suggest is adding the type to the enum, since it's (for now), as standard as every other type.

File: presto-client/src/test/java/com/facebook/presto/client/TestQueryResults.java [hunk 3]
```diff
@@ -52,6 +52,8 @@ public void testFixData()
         assertQueryResult("map(bigint,bigint)", ImmutableMap.of(1, 3, 2, 4), ImmutableMap.of(1L, 3L, 2L, 4L));
         assertQueryResult("json", "{\"json\": {\"a\": 1}}", "{\"json\": {\"a\": 1}}");
         assertQueryResult("ipaddress", "1.2.3.4", "1.2.3.4");
+        assertQueryResult("Geometry", "POINT (1.2 3.4)", "POINT (1.2 3.4)");
+        assertQueryResult("BingTile", ImmutableMap.of("x", 1, "y", 2, "zoom", 3), ImmutableMap.of("x", 1, "y", 2, "zoom", 3));
```
Comment (martint) [inline presto-client/src/test/java/com/facebook/presto/client/TestQueryResults.java]: This last case is unrelated to this change.

**Comment by mbasmanova** [comment]
@martint , thank you for review. I addressed the comments.
